File: restored-src/scripts/manga-import-preflight.ts
import { access, readFile } from 'fs/promises'
import { join } from 'path'
import { createClient } from '@libsql/client'

async function checkFile(path: string) {
  try {
    await access(path)
    return true
  } catch {
    return false
  }
}

async function main() {
  const dbUrl = process.env.SQLITE_DATABASE_URL || process.env.DATABASE_URL || `file:${join(process.cwd(), '..', 'workspace', 'novel.sqlite')}`
  const workspace = process.env.NOVEL_WORKSPACE || join(process.cwd(), 'workspace')
  const storePath = join(workspace, 'novel-store.json')

  const checks = {
    workspaceExists: await checkFile(workspace),
    storeExists: await checkFile(storePath),
    dbConfigured: Boolean(dbUrl),
    dbConnected: false,
    storeReadable: false,
    storeParsed: false,
    schemaTables: false,
  }

  let store: any = null

  if (checks.storeExists) {
    try {
      const raw = await readFile(storePath, 'utf8')
      checks.storeReadable = true
      store = JSON.parse(raw)
      checks.storeParsed = true
    } catch {
      // handled below in output
    }
  }

  if (dbUrl) {
    const client = createClient({ url: dbUrl })
    try {
      await client.execute('SELECT 1')
      checks.dbConnected = true
      const result = await client.execute(`
        SELECT name
        FROM sqlite_master
        WHERE type = 'table'
          AND name IN ('projects', 'worldbuilding', 'characters', 'outlines', 'chapters', 'reviews', 'runs')
      `)
      checks.schemaTables = result.rows.length >= 7
    } finally {
      client.close()
    }
  }

  const summary = {
    ok: checks.workspaceExists && checks.storeExists && checks.storeReadable && checks.storeParsed && checks.dbConfigured && checks.dbConnected && checks.schemaTables,
    workspace,
    storePath,
    dbUrl,
    checks,
    counts: store
      ? {
          projects: Array.isArray(store.projects) ? store.projects.length : 0,
          worldbuilding: Array.isArray(store.worldbuilding) ? store.worldbuilding.length : 0,
          characters: Array.isArray(store.characters) ? store.characters.length : 0,
          outlines: Array.isArray(store.outlines) ? store.outlines.length : 0,
          chapters: Array.isArray(store.chapters) ? store.chapters.length : 0,
          reviews: Array.isArray(store.reviews) ? store.reviews.length : 0,
          runs: Array.isArray(store.runs) ? store.runs.length : 0,
        }
      : null,
  }

  console.log(JSON.stringify(summary, null, 2))

  if (!summary.ok) {
    process.exitCode = 1
  }
}

main().catch(err => {
  console.error(err)
  process.exitCode = 1
})

File: restored-src/scripts/manga-import-preflight.py
#!/usr/bin/env python3
import json
import os
import sqlite3
from pathlib import Path


def check_file(path: Path) -> bool:
    return path.exists()


def main():
    db_url = os.environ.get('SQLITE_DATABASE_URL') or os.environ.get('DATABASE_URL') or f"file:{Path.cwd().parent / 'workspace' / 'novel.sqlite'}"
    workspace = Path(os.environ.get('NOVEL_WORKSPACE') or Path.cwd() / 'workspace')
    store_path = workspace / 'novel-store.json'

    checks = {
        'workspaceExists': check_file(workspace),
        'storeExists': check_file(store_path),
        'dbConfigured': bool(db_url),
        'dbConnected': False,
        'storeReadable': False,
        'storeParsed': False,
        'schemaTables': False,
    }

    store = None
    if checks['storeExists']:
        try:
            raw = store_path.read_text(encoding='utf-8')
            checks['storeReadable'] = True
            store = json.loads(raw)
            checks['storeParsed'] = True
        except Exception:
            pass

    if db_url:
        if db_url.startswith('file:'):
            db_path = db_url[5:]
            if '?' in db_path:
                db_path = db_path.split('?', 1)[0]
        else:
            db_path = db_url
        conn = sqlite3.connect(db_path)
        try:
            conn.execute('SELECT 1')
            checks['dbConnected'] = True
            cur = conn.execute("SELECT name FROM sqlite_master WHERE type='table' AND name IN ('projects','worldbuilding','characters','outlines','chapters','reviews','runs')")
            checks['schemaTables'] = len(cur.fetchall()) >= 7
        finally:
            conn.close()

    summary = {
        'ok': all(checks.values()),
        'workspace': str(workspace),
        'storePath': str(store_path),
        'dbUrl': db_url,
        'checks': checks,
        'counts': None if store is None else {
            'projects': len(store.get('projects', [])) if isinstance(store.get('projects', []), list) else 0,
            'worldbuilding': len(store.get('worldbuilding', [])) if isinstance(store.get('worldbuilding', []), list) else 0,
            'characters': len(store.get('characters', [])) if isinstance(store.get('characters', []), list) else 0,
            'outlines': len(store.get('outlines', [])) if isinstance(store.get('outlines', []), list) else 0,
            'chapters': len(store.get('chapters', [])) if isinstance(store.get('chapters', []), list) else 0,
            'reviews': len(store.get('reviews', [])) if isinstance(store.get('reviews', []), list) else 0,
            'runs': len(store.get('runs', [])) if isinstance(store.get('runs', []), list) else 0,
        }
    }

    print(json.dumps(summary, ensure_ascii=False, indent=2))
    if not summary['ok']:
        raise SystemExit(1)


if __name__ == '__main__':
    main()
